instrumentation/lettuce/lettuce-5.1/testing/src/main/java/io/opentelemetry/instrumentation/lettuce/v5_1/AbstractLettuceReactiveClientTest.java

```diff
@@ -0,0 +1,351 @@
+                                equalTo(
+                                    SemanticAttributes.NET_SOCK_PEER_NAME,
+                                    expectedHostAttributeValue),
+                                equalTo(SemanticAttributes.NET_SOCK_PEER_PORT, port),
+                                equalTo(SemanticAttributes.DB_SYSTEM, "redis"),
+                                equalTo(SemanticAttributes.DB_STATEMENT, "GET TESTKEY"))
+                            .hasEventsSatisfyingExactly(
+                                event -> event.hasName("redis.encode.start"),
+                                event -> event.hasName("redis.encode.end"))));
+  }
+
+  // to make sure instrumentation's chained completion stages won't interfere with user's, while
+  // still recording spans
+  @Test
+  void testGetNonExistentKeyCommand()
+      throws ExecutionException, InterruptedException, TimeoutException {
+    CompletableFuture<String> future = new CompletableFuture<>();
+    String defaultVal = "NOT THIS VALUE";
+
+    getInstrumentationExtension()
+        .runWithSpan(
+            "parent",
+            () -> {
+              reactiveCommands
+                  .get("NON_EXISTENT_KEY")
+                  .defaultIfEmpty(defaultVal)
+                  .subscribe(
+                      res ->
+                          getInstrumentationExtension()
+                              .runWithSpan(
+                                  "callback",
+                                  () -> {
+                                    assertThat(res).isEqualTo(defaultVal);
+                                    future.complete(res);
+                                  }));
+            });
+
+    assertThat(future.get(10, TimeUnit.SECONDS)).isEqualTo(defaultVal);
+
+    getInstrumentationExtension()
+        .waitAndAssertTraces(
+            trace ->
+                trace.hasSpansSatisfyingExactly(
+                    span -> span.hasName("parent").hasKind(SpanKind.INTERNAL).hasNoParent(),
+                    span ->
+                        span.hasName("GET")
+                            .hasKind(SpanKind.CLIENT)
+                            .hasAttributesSatisfyingExactly(
+                                equalTo(SemanticAttributes.NET_SOCK_PEER_ADDR, "127.0.0.1"),
+                                equalTo(
+                                    SemanticAttributes.NET_SOCK_PEER_NAME,
+                                    expectedHostAttributeValue),
+                                equalTo(SemanticAttributes.NET_SOCK_PEER_PORT, port),
+                                equalTo(SemanticAttributes.DB_SYSTEM, "redis"),
+                                equalTo(SemanticAttributes.DB_STATEMENT, "GET NON_EXISTENT_KEY"))
+                            .hasEventsSatisfyingExactly(
+                                event -> event.hasName("redis.encode.start"),
+                                event -> event.hasName("redis.encode.end")),
```

Comment:
Missing the `.hasParent(trace.getSpan(0))` for `span(1)` here (it was in the original).